packages/fxa-settings/src/pages/SigninTotpCode/index.tsx

```diff
@@ -0,0 +1,152 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import React, { useState } from 'react';
+import { useForm } from 'react-hook-form';
+import { RouteComponentProps } from '@reach/router';
+import InputText from '../../components/InputText';
+import { FtlMsg } from 'fxa-react/lib/utils';
+import { useFtlMsgResolver } from '../../models/hooks';
+import { usePageViewEvent, logViewEvent } from '../../lib/metrics';
+// import { useAlertBar } from '../../models';
+import { ReactComponent as TwoFactorImg } from './graphic_two_factor_auth.svg';
+import CardHeader from '../../components/CardHeader';
+
+// --serviceName-- is the relying party
+
+export type SigninTotpCodeProps = { serviceName?: string };
+
+type FormData = {
+  confirmationCode: string;
+};
+
+const SigninTotpCode = ({
+  serviceName,
+}: SigninTotpCodeProps & RouteComponentProps) => {
+  usePageViewEvent('signin-totp-code', {
+    entrypoint_variation: 'react',
+  });
+
+  const [securityCode, setSecurityCode] = useState<string>('');
+  const [totpErrorMessage, setTotpErrorMessage] = useState<string>('');
+  const [isFocused, setIsFocused] = useState(false);
+  // const alertBar = useAlertBar();
+  const ftlMsgResolver = useFtlMsgResolver();
+
+  const onFocusMetricsEvent = 'signin-totp-code.engage';
+
+  const onFocus = () => {
+    if (!isFocused && onFocusMetricsEvent) {
+      logViewEvent('flow', onFocusMetricsEvent, {
+        entrypoint_variation: 'react',
+      });
+      setIsFocused(true);
+    }
+  };
+
+  const { handleSubmit } = useForm<FormData>({
+    mode: 'onBlur',
+    criteriaMode: 'all',
+    defaultValues: {
+      confirmationCode: '',
+    },
+  });
+
+  const onSubmit = () => {
+    if (!securityCode) {
+      const codeRequiredError = ftlMsgResolver.getMsg(
+        'signin-totp-code-required-error',
+        'Two-step authentication code required'
+      );
+      setTotpErrorMessage(codeRequiredError);
+    }
+    try {
+      // Check security code
+      // Log success event
+      // Check if isForcePasswordChange
+    } catch (e) {
+      // TODO: error handling, error message confirmation
+      // const errorSigninTotpCode = ftlMsgResolver.getMsg(
+      //   'signin-totp-code-error-general',
+      //   'Invalid confirmation code'
+      // );
+      // alertBar.error(errorSigninTotpCode);
+    }
+  };
+
+  return (
+    // TODO: redirect to force_auth or signin if user has not initiated sign in
+    <>
+      <CardHeader
+        headingWithDefaultServiceFtlId="signin-totp-code-heading-w-default-service"
+        headingWithCustomServiceFtlId="signin-totp-code-heading-w-custom-service"
+        headingText="Enter security code"
+        {...{ serviceName }}
+      />
+
+      <main>
+        <div className="flex justify-center mx-auto">
+          <TwoFactorImg className="w-3/5" role="img" />
```

Comment:
All `role="img"` SVGs should be given `aria-label`s which will work like alt text. If content-server doesn't provide it, we can make our best guess on what describes it and if we're not sure we can run it by Vesta. Maybe this can be, "Device with hidden code" or something like that.

Reply:
@LZoog I feel this image is decorative (doesn't add information to the page) and doesn't need alt text, as described here: https://www.w3.org/WAI/tutorials/images/decorative/
Does the team prefer to add alt-text to all images, decorative or not? Also, I understand that wheter or not this image is decorative might be subjective.